app/src/main/java/org/sopt/and/presentation/main/MainScreen.kt

```diff
@@ -0,0 +1,109 @@
+package org.sopt.and.presentation.main
+
+import androidx.compose.animation.EnterTransition
+import androidx.compose.animation.ExitTransition
+import androidx.compose.foundation.background
+import androidx.compose.foundation.layout.PaddingValues
+import androidx.compose.foundation.layout.padding
+import androidx.compose.foundation.layout.statusBarsPadding
+import androidx.compose.foundation.layout.systemBarsPadding
+import androidx.compose.material3.Scaffold
+import androidx.compose.runtime.Composable
+import androidx.compose.ui.Modifier
+import androidx.navigation.NavHostController
+import androidx.navigation.compose.NavHost
+import org.sopt.and.core.designsystem.theme.WavveBackground
+import org.sopt.and.core.navigation.Route
+import org.sopt.and.core.preference.PreferenceUtil.Companion.LocalPreference
+import org.sopt.and.presentation.home.navigation.Home
+import org.sopt.and.presentation.home.navigation.homeScreen
+import org.sopt.and.presentation.main.component.MainBottomBar
+import org.sopt.and.presentation.mypage.navigation.myPageScreen
+import org.sopt.and.presentation.search.navigation.searchScreen
+import org.sopt.and.presentation.signin.navigation.SignIn
+import org.sopt.and.presentation.signin.navigation.signInScreen
+import org.sopt.and.presentation.signup.navigation.signUpScreen
+
+@Composable
+fun MainScreen(
+    mainNavigator: MainNavigator = rememberMainNavigator()
+) {
+    Scaffold(
+        bottomBar = {
+            MainBottomBar(
+                isVisible = mainNavigator.showBottomBar(),
+                tabs = MainTab.entries,
+                currentTab = mainNavigator.currentTab,
+                onTabSelected = mainNavigator::navigate
+            )
+        }
+    ) { paddingValues ->
+        val startDestination = getStartDestination()
+
+        MainNavHost(
+            navController = mainNavigator.navController,
+            startDestination = startDestination,
+            paddingValues = paddingValues
+        )
+    }
+}
+
+@Composable
+private fun getStartDestination(): Route {
+    val preference = LocalPreference.current
+    return if (preference.id.isEmpty() || preference.password.isEmpty()) {
+        SignIn
+    } else {
+        Home
+    }
+}
```

Comment:
```suggestion
@Composable
private fun getStartDestination(): Route = 
    if (LocalPreference.current.run { id.isEmpty() || password.isEmpty() }) SignIn else Home
```
scope 함수를 사용하면 더 간결하게 적을 수 있을 것 같아요.